osu.Game/Rulesets/Scoring/HealthProcessor.cs

```diff
@@ -31,16 +34,24 @@ public abstract partial class HealthProcessor : JudgementProcessor
         /// </summary>
         public bool HasFailed { get; private set; }
 
+        /// <summary>
+        /// List of triggers that caused fail
+        /// </summary>
+        public List<IHasFailCondition> FailTriggers = new List<IHasFailCondition>();
+
         /// <summary>
         /// Immediately triggers a failure for this HealthProcessor.
         /// </summary>
-        public void TriggerFailure()
+        public void TriggerFailure(IHasFailCondition? Trigger = null)
```

Comment:
Probably make this accept a `Mod` instead, call it `triggeringMod` and add documentation.

Reply:
Right